doc/lua_api.txt

```diff
@@ -4428,8 +4428,11 @@ Callbacks:
     * Called when the object is instantiated.
     * `dtime_s` is the time passed since the object was unloaded, which can be
       used for updating the entity state.
-* `on_deactivate(self)
+* `on_deactivate(self, removal)
     * Called when the object is about to get removed or unloaded.
+	* `removal`: boolean indicating whether the object is about to get removed.
+	* Note that this won't be called if the object hasn't been activated in the first place.
+	  In particular, `minetest.clear_objects({mode = "full"})` won't call this.
```

Comment:
> Surely clearobjects quick will not activate objects and then immediately deactivate them but get rid of them immediately, making this statement wrong.

Doesn't "clearobjects quick" basically just iterate over active entities, deactivating them?

Reply:
Then it would not fulfill its purpose of clearing objects on the entire map.
The TL;DR is that when loading a block the engine knows whether it needs to throw away all objects in it and `on_deactivate` obviously won't be called in that case.

Reply:
> Then it would not fulfill its purpose of clearing objects on the entire map.

"clearobjects quick" doesn't clear the entire map, but indeed:

> Clear objects immediately in loaded mapblocks, clear objects in unloaded mapblocks only when the mapblocks are next activated.

Meaning `on_deactivate` would be skipped for those.

Reply:
So do you agree that your example is wrong? clearobjects full with definitely not call the callback while clearobjects quick *might* call the callback.

Reply:
> So do you agree that your example is wrong?

No, my example is correct (clearobjects full *doesn't* call the callback). It may be misleading though, as the reader may think that it is implied that clearobjects quick always calls this.

> clearobjects quick might call the callback.

I should add such a line.